packages/fxa-auth-server/scripts/audit-tokens.ts

```diff
@@ -0,0 +1,519 @@
+  accountResetTokens: toTable('accountResetTokens'),
+  accounts: toTable('accounts'),
+  devices: toTable('devices'),
+  deviceCommands: toTable('deviceCommands'),
+  emails: toTable('emails'),
+  keyFetchTokens: toTable('keyFetchTokens'),
+  linkedAccounts: toTable('linkedAccounts'),
+  passwordChangeTokens: toTable('passwordChangeTokens'),
+  passwordForgotTokens: toTable('passwordForgotTokens'),
+  paypalCustomers: toTable('paypalCustomers'),
+  recoveryCodes: toTable('recoveryCodes'),
+  recoveryKeys: toTable('recoveryCodes'),
+  securityEvents: toTable('securityEvents'),
+  sentEmails: toTable('sentEmails'),
+  sessionTokens: toTable('sessionTokens'),
+  signinCodes: toTable('signinCodes'),
+  totp: toTable('totp'),
+  unblockCodes: toTable('unblockCodes'),
+  unverifiedTokens: toTable('unverifiedTokens'),
+  verificationReminders: toTable('verificationReminders'),
+  clientDevelopers: toTable('clientDevelopers', 'fxa_oauth'),
+  clients: toTable('clients', 'fxa_oauth'),
+  codes: toTable('codes', 'fxa_oauth'),
+  developers: toTable('developers', 'fxa_oauth'),
+  refreshTokens: toTable('refreshTokens', 'fxa_oauth'),
+  tokens: toTable('tokens', 'fxa_oauth'),
+  profile: toTable('profile', 'fxa_profile'),
+};
+
+//#endregion
+
+//#region Result Handling
+function formatStatLabel(label: string) {
+  let cleaned = label;
+  cleaned = cleaned.replace(/(fxa|fxa_profile|fxa_oauth)\./g, '$1_');
+  return cleaned;
+}
+
+function formatResult(result: any) {
+  let formatted: any;
+
+  // For key value pairs pivot the result
+  if (result?.[0]?.[0]?.group_key && result?.[0]?.[0]?.group_value) {
+    const pivotResult: any = {};
+    for (const row of result?.[0] || []) {
+      pivotResult[row.group_key] = row.group_value;
+    }
+    formatted = pivotResult;
+  } else {
+    // Return the first row
+    formatted = result?.[0]?.[0];
+  }
+
+  return formatted;
+}
+
+export function emitStats(name: string, result: any) {
+  if (!result) {
+    return;
+  }
+
+  Object.entries(result).forEach(([key, val]) => {
+    if (typeof val === 'number') {
+      const label = formatStatLabel(`db-audit.${name}.${key}`);
+      statsd.gauge(label, val);
+      log.debug('emit-stats', { label });
+    }
+  });
+}
+
+export function logResult(name: string, query: string, result: any) {
+  log.info('result', { name, result });
+
+  if (program.verbose) {
+    function resultSummary() {
+      if (result) {
+        const pairs = Object.entries(result)
+          .map(([k, v]) => `\n-- ${k}: ${v}`)
+          .join('');
+
+        return `\n-- RESULT SUMMARY:\n${pairs}`;
+      }
+
+      return '-- RESULT SUMMARY: No Result';
+    }
+    console.log(
+      `\n-- AUDIT: ${name}\n-- QUERY:\n${query}\n${resultSummary()}\n\n\n`
+    );
+  }
+}
+//#endregion
+
+//#region Audits
+
+/** Helper function to determine the number of rows to query. */
+function getSampleSize(_tableName: string) {
+  // We probably are fine just using a large number for sample size. Down
```

Comment:
famous last words, and so on 😅

Reply:
@chenba Do you have any thoughts on this by the way. I am sort of gambling on the fact the data is pretty randomly distributed due to how we generate ids. What are you thoughts on this? I considered trying to find the standard deviation so we could actually calculate a min sampling size, but I shelved for another time. I doubt it will be necessary.

Reply:
> data is pretty randomly distributed due to how we generate ids

Sadly uuids are not uniformly distributed.  But I agree that let's see how well a "large" number work first, and then go from there.